src/common/remove_all_invalid_leading_chars.rs

```diff
@@ -0,0 +1,18 @@
+pub fn remove_all_invalid_leading_chars(string: &str) -> String {
+    let mut cleaned_string = string;
+
+    while !cleaned_string.starts_with(|c: char| c.is_alphabetic() || c == '_') {
```

Comment:
I would recommend using the `skip_while` iterator, prevents the need to copy after each skipped char.

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.skip_while

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=acd50e88e87a367055f4da35e46febda

Reply:
Thank you! I did not know about that function.